MvvmCross/ViewModels/Result/IMvxResultAwaitingViewModel.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MS-PL license.
+// See the LICENSE file in the project root for more information.
+
+namespace MvvmCross.ViewModels.Result;
+
+/// <summary>
+/// Implementing this interface requires to manually register and unregister ViewModel to result:
+/// <code>
+/// protected override void ReloadFromBundle(IMvxBundle state)
+/// {
+///     base.ReloadFromBundle(state);
+///     this.ReloadAndRegisterToResult&lt;<typeparamref name="TResult"/>&gt;(state);
+/// }
+/// protected override void SaveStateToBundle(IMvxBundle bundle)
+/// {
+///     base.SaveStateToBundle(bundle);
+///     this.SaveRegisterToResult&lt;<typeparamref name="TResult"/>&gt;(bundle);
+/// }
+/// public override ViewDestroy(bool viewFinishing = true)
+/// {
+///     base.ViewDestroy();
+///     if (viewFinishing)
+///         this.UnregisterToResult&lt;<typeparamref name="TResult"/>&gt;();
+/// }
+/// </code>
+/// </summary>
+/// <typeparam name="TResult"></typeparam>
+public interface IMvxResultAwaitingViewModel<TResult> : IMvxViewModel
```

Comment:
That is needed to register VMs in `MvxResultViewModelManager._registrations`. Probably not the best way. I could make non-generic `IMvxResultAwaitingViewModel` that `IMvxResultAwaitingViewModel<TResult>` inherits from and use the same in `MvxResultViewModelManager._registrations` instead of `IMvxViewModel`. But then it's kind of incorrect since `IMvxResultAwaitingViewModel` supposes to have a result to await. End users may be confused which to use.
On the other hand I could use `object` in `MvxResultViewModelManager._registrations` instead of `IMvxViewModel`

Reply:
Maybe instead of using `IMvxViewModel` as a common base, then use a new interface. Then you have something for `MvxResultViewModelManager` to look for, while not depending on `IMvxViewModel`.